x/auth/vesting/types/vesting_account.go

```diff
@@ -570,3 +570,69 @@ func (dva DelayedVestingAccount) String() string {
 	out, _ := dva.MarshalYAML()
 	return out.(string)
 }
+
+//-----------------------------------------------------------------------------
+// Permanent Locked Vesting Account
+
+var _ vestexported.VestingAccount = (*PermanentLockedVestingAccount)(nil)
+var _ authtypes.GenesisAccount = (*PermanentLockedVestingAccount)(nil)
+
+// NewPermanentLockedVestingAccountRaw creates a new PermanentLockedVestingAccount object from BaseVestingAccount
+func NewPermanentLockedVestingAccountRaw(bva *BaseVestingAccount) *PermanentLockedVestingAccount {
+	// ensure EndTime is set to -1, as PermanentLockedVestingAccount's do not have an EndTime
+	bva.EndTime = -1
```

Comment:
Seems kind of hacky... but otherwise we may unwillingly store an `EndTime` value from an inherited BaseVestingAccount. This `EndTime` doesn't actually do anything... as all the logic is handled in the implementations of the `PermanentLockedVestingAccount`'s functions.

Reply:
Why do we need to embed BaseVestingAccount at all? If we don't embed we won't have this problem

Reply:
Even if we drop the embedded `BaseVestingAccount` we will need to implement `GetStartTime()` and `GetEndTime()` somehow to implement the exported VestingAccount interface [see here](https://github.com/cosmos/cosmos-sdk/blob/master/x/auth/vesting/exported/exported.go#L35-L36).

@aaronc were you suggesting we refactor that interface as well? Or a strategy where we actually don't consider `PermanentLockedVestingAccount` a vesting account at all? I guess there isn't really anything "vesting" about it. So maybe decoupling it entirely from the vesting account interface is prefereble... I'd have to investigate what that would look like.

Reply:
Let's keep it simple for now IMHO even if we have to do something weird for the interface. But we should check consequences on returning invalidate times. When we get around to it I want to refactor vesting accounts into something much more modular.

Reply:
> Can it embed BaseAccount, + another OriginalVesting field?

My suggestion above would require duplicating a lot of the `BaseVestingAccount` logic into `PermanentLockedVestingAccount`, so maybe not the best idea.

I think @clevinson what you're doing here seems like the most straightforward way right now. Internally in the SDK, we don't use `GetEndTime` ever (except in the local ValidateBasic). If a user queries this account (via gRPC or whatever), then returning `-1` on that field should be pretty clear that it doesn't have an EndTime.